src/Orleans/Serialization/OrleansJsonSerializer.cs

```diff
@@ -237,5 +233,29 @@ public override object ReadJson(JsonReader reader, Type objectType, object exist
         }
     }
 
+    class GuidConverter : JsonConverter
+    {
+        public override bool CanConvert(Type objectType)
+        {
+            return objectType.IsAssignableFrom(typeof(Guid));
```

Comment:
This is the implementation I've been using, by the way: https://gist.github.com/ReubenBond/9416e7a4c528eab473fd